src/SourceGenerators/Uno.UI.Tasks/RuntimeAssetsSelector/RuntimeAssetsSelectorTask.cs

```diff
@@ -0,0 +1,115 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Http.Headers;
+using System.Runtime.InteropServices.ComTypes;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading;
+using System.Transactions;
+using System.Xml;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Mono.Cecil;
+
+namespace Uno.UI.Tasks.RuntimeAssetsSelector
+{
+	/// <summary>
+	/// A task used to merge linker definition files and embed the result in an assembly
+	/// </summary>
+	public class RuntimeAssetsSelectorTask_v0 : Microsoft.Build.Utilities.Task
+	{
+		[Required]
+		public Microsoft.Build.Framework.ITaskItem[]? UnoRuntimeEnabledPackage { get; set; }
+
+		[Required]
+		public string UnoRuntimeIdentifier { get; set; } = "";
+
+		public string TargetFrameworkVersion { get; set; } = "";
+
+		[Output]
+		public Microsoft.Build.Framework.ITaskItem[]? Assemblies { get; set; }
+
+		[Output]
+		public Microsoft.Build.Framework.ITaskItem[]? DebugSymbols { get; set; }
+
+		public override bool Execute()
+		{
+			try
+			{
+				List<ITaskItem> assemblies = new();
+				List<ITaskItem> debugSymbols = new();
+
+				foreach (var package in UnoRuntimeEnabledPackage ?? Array.Empty<ITaskItem>())
+				{
+					var packageBasePath = package.GetMetadata("PackageBasePath");
+
+					if (!Version.TryParse(TargetFrameworkVersion.Substring(1), out var targetFrameworkVersion))
```

Comment:
If the `TargetFrameworkVersion` is not set `""` this will throw, right?

Reply:
It's based on a trick where `public string TargetFrameworkVersion { get; set; } = "";` is the default, and that `TargetFrameworkVersion ` is not set when empty from msbuild. So it will never be set to null. It's also not marked as required because netstandard2.0 does not set this property.